src/Shared/OpportunisticIntern.cs

```diff
@@ -972,6 +984,13 @@ internal void ReportStatistics(string heading)
                             return false;
                         }
                     }
+#if !CLR2COMPATIBILITY
+                    else if (_useSimpleConcurrency)
+                    {
+                        interned = _internedStrings.GetOrAdd(candidate.ExpensiveConvertToString(), candidate2 => candidate2);
```

Comment:
I wonder if it'd be cheaper to avoid the `valueFactory` overload:

```c#
string stringified = candidate.ExpensiveConvertToString();
interned = _internedStrings.GetOrAdd(stringified, stringified);
```